webapp/src/components/surveyDateSelector/surveyDateSelector.tsx

```diff
@@ -1,36 +1,41 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useCallback} from 'react';
-import {format} from 'date-fns';
+import React, {useEffect, useState} from 'react';
 
 import DatePicker from 'components/common/datePicker';
 
 import './style.scss';
 
+const dateFormattingOptions: Intl.DateTimeFormatOptions = {
+    year: 'numeric',
+    month: '2-digit',
+    day: 'numeric',
+};
+
 export type Props = {
-    value: Date
-    onChange?: (value: Date) => void
+    value: Date;
+    onChange: (value: Date) => void;
 };
 
 const SurveyDateSelector = ({value, onChange}: Props) => {
-    const onSelectHandler = useCallback((newValue: Date) => {
-        if (onChange) {
-            onChange(newValue);
-        }
-    }, [onChange]);
+    const [formatedDate, setFormattedDate] = useState<string>('');
+
+    useEffect(() => {
+        setFormattedDate(value.toLocaleDateString(undefined, dateFormattingOptions));
+    }, [value]);
```

Comment:
This causes an extra re-render every time the prop changes. Unless there are reasons to perform that extra re-render, you can change this to a `useMemo`.

Also, not sure how expensive is `toLocalDateString`, but if it is cheap, we could remove the useMemo also.

Reply:
yeah, `useMemo` makes more sense.

Reply:
Done.